cs50/2017/fall/recover/check50/__init__.py

```diff
@@ -72,25 +49,57 @@ def test_noimage(self):
     @check("compiles")
     def first_image(self):
         """recovers 000.jpg correctly"""
+        self.initialize_data()
         self.add("card.raw")
         self.spawn("./recover card.raw").exit(0, timeout=10)
-        if self.hash("000.jpg") != self.hashes[0]:
-            raise Error("recovered image does not match")
+        self.check_jpg("000.jpg")
 
     @check("compiles")
     def middle_images(self):
         """recovers middle images correctly"""
+        self.initialize_data()
         self.add("card.raw")
         self.spawn("./recover card.raw").exit(0, timeout=10)
-
-        for i, hash in enumerate(self.hashes[1:-1], 1):
-            if hash != self.hash("{:03d}.jpg".format(i)):
-                raise Error("recovered image does not match")
+        for i in range(1, 49):
+            self.check_jpg("{:03d}.jpg".format(i))
 
     @check("compiles")
     def last_image(self):
         """recovers 049.jpg correctly"""
+        self.initialize_data()
         self.add("card.raw")
         self.spawn("./recover card.raw").exit(0, timeout=10)
-        if self.hash("049.jpg") != self.hashes[-1]:
-            raise Error("recovered image does not match")
+        self.check_jpg("049.jpg")
+
+    def check_jpg(self, jpg):
+        """Checks for a jpeg to have the correct hash, and if it does not,
+        raises and error with a helpful hint"""
```

Comment:
For multiline method docstrings, I think convention might be to put \n after first """ and before last """?